spec/factories.rb

```diff
@@ -11,4 +18,29 @@
       FakeResponse.new(body || payload.to_json)
     end
   end
+
+  factory :schema, class: FakeSchema do
+    sequence(:name) { |n| "json_schema-#{n}" }
+
+    trait :invalid do
+      json { "" }
+    end
+
+    skip_create
+
+    initialize_with do
+      name = attributes.fetch(:name)
+      json_attribute = attributes.fetch(:json, nil)
+      attributes_as_json = attributes.except(:json, :name)
+
+      FakeSchema.new(name, json_attribute || attributes_as_json)
+    end
+
+    after :create do |schema|
+      path = File.join(JsonMatchers.schema_root, "#{schema.name}.json")
+      payload = JsonMatchers::Payload.new(schema.json)
+
+      IO.write(path, payload)
```

Comment:
@mike-burns thanks for providing an alternate implementation in https://github.com/thoughtbot/json_matchers/pull/69#issuecomment-373800142, I've incorporated some of those ideas here (mainly to use `IO.write` instead of File.open`)!